src/Microsoft.Dnx.Compilation.CSharp/RoslynCompiler.cs

```diff
@@ -252,58 +252,43 @@ private CompilationModules GetCompileModules(CompilationTarget target)
             });
         }
 
-        private static CSharpCompilation ApplyVersionInfo(CSharpCompilation compilation, CompilationProjectContext project,
+        private static CSharpCompilation ApplyProjectInfo(CSharpCompilation compilation, CompilationProjectContext project,
             CSharpParseOptions parseOptions)
         {
-            const string assemblyFileVersionName = "System.Reflection.AssemblyFileVersionAttribute";
-            const string assemblyVersionName = "System.Reflection.AssemblyVersionAttribute";
-            const string assemblyInformationalVersion = "System.Reflection.AssemblyInformationalVersionAttribute";
+            var attributeNameToProjectValue = new Dictionary<string, string>(StringComparer.Ordinal)
+            {
+                ["System.Reflection.AssemblyTitleAttribute"] = project.Title,
+                ["System.Reflection.AssemblyDescriptionAttribute"] = project.Description,
+                ["System.Reflection.AssemblyCopyrightAttribute"] = project.Copyright,
+                ["System.Reflection.AssemblyFileVersionAttribute"] = project.AssemblyFileVersion.ToString(),
+                ["System.Reflection.AssemblyVersionAttribute"] = RemovePrereleaseTag(project.Version),
+                ["System.Reflection.AssemblyInformationalVersionAttribute"] = project.Version
+            };
 
             var assemblyAttributes = compilation.Assembly.GetAttributes();
 
-            var foundAssemblyFileVersion = false;
-            var foundAssemblyVersion = false;
-            var foundAssemblyInformationalVersion = false;
-
-            foreach (var assembly in assemblyAttributes)
+            foreach (var attribute in assemblyAttributes)
             {
-                string attributeName = assembly.AttributeClass.ToString();
+                string attributeName = attribute.AttributeClass.ToString();
 
-                if (string.Equals(attributeName, assemblyFileVersionName, StringComparison.Ordinal))
-                {
-                    foundAssemblyFileVersion = true;
-                }
-                else if (string.Equals(attributeName, assemblyVersionName, StringComparison.Ordinal))
-                {
-                    foundAssemblyVersion = true;
-                }
-                else if (string.Equals(attributeName, assemblyInformationalVersion, StringComparison.Ordinal))
+                if (attributeNameToProjectValue.ContainsKey(attributeName))
                 {
-                    foundAssemblyInformationalVersion = true;
+                    attributeNameToProjectValue.Remove(attributeName);
```

Comment:
Nit: You could build the attributes string right here, no need to iterate over the remaining collection one more time.